### PR TITLE
feat: add setAutoRefreshInterval method to xandr controller

### PR DESCRIPTION
**Pull request by andriensis**
## Description

- Add setAutoRefreshInterval to `XandrController`
- Fix `autoRefreshInterval` value on banners

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [x] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/xandr/lib/ad_banner.dart [hunk 1]
```diff
@@ -47,6 +47,11 @@ class AdBanner extends StatefulWidget {
               (allowNativeDemand == true && nativeAdBuilder != null),
           'nativeAdBuilder must be set if allowNativeDemand is true',
         ),
+        assert(
```
Comment (thekorn) [inline packages/xandr/lib/ad_banner.dart:50]: Can you please add a docstring here, referencing the documentation, and maybe quoting the relevant lines?
```suggestion
      //opensdk:auto_refresh_interval or adview.setAutoRefreshInterval(long interval): ....
      // see: https://learn.microsoft.com/en-us/xandr/mobile-sdk/show-banners-on-android
        assert(
```

File: packages/xandr/lib/ad_banner.dart [hunk 2]
```diff
@@ -376,7 +381,7 @@ class _HostAdBannerView extends StatelessWidget {
         creationParams = <String, dynamic>{
           'placementID': placementID,
           'inventoryCode': inventoryCode,
-          'autoRefreshInterval': autoRefreshInterval.inMilliseconds,
+          'autoRefreshInterval': autoRefreshInterval.inSeconds,
```
Comment (thekorn) [inline packages/xandr/lib/ad_banner.dart:384]: im confused, in the docs they say "in milliseconds" - woooot?

File: packages/xandr_android/lib/xandr_android.dart [hunk 3]
```diff
@@ -162,6 +162,19 @@ class XandrAndroid extends XandrPlatform {
...
     return _api.setGDPRPurposeConsents(purposeConsents);
   }
+
+  @override
+  Future<bool> setAutoRefreshInterval(
+    int autoRefreshIntervalInSeconds,
+    String? inventoryCode,
+    String? placementID,
+  ) {
+    return _api.setAutoRefreshInterval(
```
Comment (thekorn) [inline packages/xandr_android/lib/xandr_android.dart:172]: we should assert here that exactly one of inventory code or placement must be !=null
(to not endup in the "XandrBanner for inventoryCode=null, placementID=null not found!" error case in getBannerWithCode())
same for the iOS equivalent

File: packages/xandr_ios/lib/xandr_ios.dart [hunk 4]
```diff
@@ -162,6 +162,19 @@ class XandrIOS extends XandrPlatform {
...
     return _api.setGDPRPurposeConsents(purposeConsents);
   }
+
+  @override
+  Future<bool> setAutoRefreshInterval(
+    int autoRefreshIntervalInSeconds,
+    String? inventoryCode,
+    String? placementID,
+  ) {
+    return _api.setAutoRefreshInterval(
```
Comment (thekorn) [inline packages/xandr_ios/lib/xandr_ios.dart:172]: please assert, see android case

File: packages/xandr_platform_interface/lib/src/interface.dart [hunk 5]
```diff
@@ -184,4 +184,23 @@ abstract class XandrPlatform extends PlatformInterface {
...
+  /// [inventoryCode] is the optional banner inventory code
+  /// [placementID] is the optional banner placement ID
+  /// At least one of [inventoryCode] or [placementID] must be filled
+  /// to identify the banner
+  Future<bool> setAutoRefreshInterval(
+    int autoRefreshIntervalInSeconds,
+    String? inventoryCode,
+    String? placementID,
+  ) {
+    return _instance.setAutoRefreshInterval(
```
Comment (thekorn) [inline packages/xandr_platform_interface/lib/src/interface.dart:200]: technically please assert here as well - although this method channel implementation is/would only be used in tests.....

**Review by thekorn — CHANGES_REQUESTED** [review]

**Comment by thekorn** [comment]
thanks @andriensis - i added a few suggestions, please let me know if you want to discuss any of those

File: packages/xandr/lib/ad_banner.dart (hunk 2 above)
Comment (andriensis) [reply packages/xandr/lib/ad_banner.dart:384]: yeah, this should be in seconds
look here for instance: https://learn.microsoft.com/en-us/xandr/mobile-sdk/show-banners-on-ios?tabs=swift1

`banner.autoRefreshInterval = 60`

File: packages/xandr_android/lib/xandr_android.dart (hunk 3 above)
Comment (thekorn) [reply packages/xandr_android/lib/xandr_android.dart:172]: thats incorrect, sorry for the noise,

File: packages/xandr_ios/lib/xandr_ios.dart (hunk 4 above)
Comment (thekorn) [reply packages/xandr_ios/lib/xandr_ios.dart:172]: please also ignore

**Pull request merged**